components/script/dom/readablestream.rs

```diff
@@ -403,16 +406,18 @@ impl ReadableStream {
     /// must be done before `read_a_chunk`.
     /// Native call to
     /// <https://streams.spec.whatwg.org/#acquire-readable-stream-reader>
-    pub fn start_reading(&self) -> Result<(), ()> {
+    pub fn acquire_default_reader(
+        &self,
+        can_gc: CanGc,
+    ) -> Fallible<DomRoot<ReadableStreamDefaultReader>> {
         // Let reader be a new ReadableStreamDefaultReader.
-        // Perform ? SetUpReadableStreamDefaultReader(reader, stream).
-        // Return reader.
-        let reader = ReadableStreamDefaultReader::set_up(&self.global(), self, CanGc::note())
-            .map_err(|_| ())?;
+        let reader = ReadableStreamDefaultReader::new_inherited(&self.global(), can_gc);
```

Comment:
I think the crash is actually a memory error disguised as a borrow error, because here the reader is not rooted yet and yet the below call can gc(and also clears the read requets vec).

Reply:
The crash at `add_read_request` goes away if you root the reader before setting it up, but there is another borrow error now at `get_num_read_requests`, looking into it...

Reply:
Ok the second problem was the same thing, but in the constructor, see: https://github.com/gterzian/servo/commit/17411f01b14086a14424680a5e0658fc81a8ce84